machine-learning-scikit-learn.ipynb

```diff
@@ -0,0 +1,333 @@
+   "outputs": [],
+   "source": [
+    "from dask_ml.wrappers import Incremental\n",
+    "\n",
+    "inc = Incremental(est, scoring='accuracy')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our model (SGDClassifier here) must implement `partial_fit` to be used with `Incremental`. A list of models that implement this API is available on a scikit-learn documentation page under \"Incremental learning\": http://scikit-learn.org/stable/modules/scaling_strategies.html#incremental-learning\n",
+    "\n",
+    "`Incremental` does data management: it calls `est.partial_fit` on each chunk of the passed data. Dask moves the model to different chunks of the data (or vice versa) to let this happen. Note: Dask-ML `Incremental` gives data to Scikit-learn and does not change the underlying optimization algorithm that Scikit-learn uses. There are different optimization algorithms in Dask-ML that do this, especially for clustering and matrix decomposition.\n",
+    "\n",
+    "It is important to specify the scoring parameter in `Incremental`; otherwise, scikit-learn scorers are fed Dask arrays (which they're not optimized for).\n",
+    "\n",
+    "\n",
+    "[1]:http://scikit-learn.org/stable/modules/scaling_strategies.html\n",
+    "[inc]:http://scikit-learn.org/stable/modules/scaling_strategies.html#incremental-learning"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model training\n",
+    "For an `Incremental` model, `partial_fit` and `fit` aliased to the same item and do one complete pass over the dataset. Let's call `partial_fit` a couple times, and score it after it's called each time (which will allow for a visualization):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "# takes about 1min\n",
+    "\n",
+    "from distributed.metrics import time\n",
+    "data = []\n",
+    "start = time()\n",
+    "for iteration in range(10):\n",
+    "    inc.partial_fit(X_train, y_train, classes=da.unique(y))\n",
+    "    data += [{'score': inc.score(X_test, y_test),\n",
+    "             'epochs': iteration + 1,\n",
+    "             'time': time() - start}]\n",
+    "    print(data[-1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "df = pd.DataFrame(data)\n",
+    "df.plot(x='epochs', y='score')\n",
+    "plt.title('Score over iterations')\n",
+    "plt.ylabel('Score')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Predicting\n",
+    "Let's predict, than compute the accuracy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_hat = inc.predict(X_test)\n",
```

Comment:
I think scikit-learn uses `y_pred`, so people may be more familiar with that (though I'm partial to `y_hat`, in this case it's best to go where the people are).